src/elf/note.rs

```diff
@@ -205,7 +205,7 @@ if_alloc! {
             };
             debug!("{:?} - {:#x}", header, *offset);
             // -1 because includes \0 terminator
-            let name = bytes.gread_with::<&'a str>(offset, ctx::StrCtx::Length(header.n_namesz - 1))?;
+            let name = bytes.gread_with::<&'a str>(offset, ctx::StrCtx::Length(header.n_namesz.saturating_sub(1)))?;
```

Comment:
won't this read a zero-length name? What does this mean exactly, that the note is empty itself, or that it doesn't have a note name, just a desc? I'm not super familiar with elf notes lately, and what the expectations are (probably none?)

Reply:
I got a crash in goblin where when reaching here the value of header is:
```
goblin::elf::note::NoteHeader {n_namesz: 0, n_descsz: 0, n_type: 0}
```
I suppose this means the not section is empty.

Reply:
What does gread_with do when given a length of 0?

Reply:
i don't know offhand :sweat_smile: ; i suspect it'll just return an empty string, let me check

Reply:
it does this (test passes):
```rust
#[test]
fn pread_zero_length() {
    let bytes = b"hello";
    let s = bytes.pread_with::<&str>(0, ctx::StrCtx::Length(0));
    assert!(s.is_ok());
    assert_eq!(s.unwrap(), "");
}
```

Reply:
my guess is that gread_with increases offset with the length read, so the *offset += 1 is to go past the \0 terminator when there is one to continue reading the desc.

Reply:
yes i think this is correct, assuming an empty note is `\0` in the bytes (which sort of makes sense to me); are you expecting something in desc? if this fixes it for you we can just add it; it won't hurt anything and it's certainly better than panicking. my experience with notes is they're just kind of a super unspecified thing, hah. if someone comes and complains that they have an empty note name but their desc is off by 1, then they weren't using this lib before since it would have panicked, and we can just cross that bridge or fix it when it happens